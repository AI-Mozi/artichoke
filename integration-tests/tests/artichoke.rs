use integration_tests::run;

const BINARY: &str = "artichoke";
const FIXTURES_ROOT: &str = "./tests/fixtures/";

#[cfg(target_family = "windows")]
#[test]
fn help_windows() {
    insta::assert_toml_snapshot!(run(BINARY, &["--help"]).unwrap());
}

#[cfg(target_family = "unix")]
#[test]
fn help_unix() {
    insta::assert_toml_snapshot!(run(BINARY, &["--help"]).unwrap());
}

#[test]
fn hello_world() {
    let app_name = "hello_world.rb";
    let path = format!("{}{}", FIXTURES_ROOT, app_name);
    insta::assert_toml_snapshot!(run(BINARY, &[&path]).unwrap());
}

#[test]
fn fizz_buzz() {
    let app_name = "fizz_buzz.rb";
    let path = format!("{}{}", FIXTURES_ROOT, app_name);
    insta::assert_toml_snapshot!(run(BINARY, &[&path]).unwrap());
}
