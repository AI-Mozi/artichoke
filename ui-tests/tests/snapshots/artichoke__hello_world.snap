---
source: tests/artichoke.rs
assertion_line: 22
expression: "run(BINARY, &[&path]).unwrap()"

---
call_args = ['./tests/fixtures/hello_world.rb']
status = 0
stdout = '"Hello Artichoke!"'
stderr = ''
